api/repositories/service_binding_repository.go

```diff
@@ -75,6 +77,12 @@ func (r ServiceBindingRecord) Relationships() map[string]string {
 	}
 }
 
+type Credentials map[string]any
+type ServiceBindingDetailsRecord struct {
+	Credentials    Credentials
+	SyslogDrainURL *string
```

Comment:
Provided that the repository never populates `SyslogDrainURL` and `VolumeMounts`, I would suggest removing them from the record. We can introduce them when we properly implement them.